.github/workflows/regression.yml

```diff
@@ -192,12 +192,31 @@ jobs:
           done
           find $BIN_FOLDER -type f -exec chmod a+x '{}' \;
           ls -lahR $BIN_FOLDER
+          echo "cloning branch"
+          branch="${{github.ref}}"
+          name="SU2"
+          SRC_FOLDER="$PWD/src"
+          mkdir -p $SRC_FOLDER 
+          cd $SRC_FOLDER 
+          git clone --recursive --depth=1 --shallow-submodules https://github.com/su2code/SU2 $name
+          cd $name
+          git config --add remote.origin.fetch '+refs/pull/*/merge:refs/remotes/origin/refs/pull/*/merge'
+          git config --add remote.origin.fetch '+refs/heads/*:refs/remotes/origin/refs/heads/*'
+          git fetch origin --depth=1 $branch:$branch
+          git checkout $branch
+          git submodule update
+          echo $PWD
+          ls -lahR
+          cd ..
+          echo "done cloning"
+          echo $PWD
+          ls -lahR
```

Comment:
This needs some additional review. Copied from CompileSU2.sh (docker-build repository)

Reply:
we keep it like this as discussed in the dev meeting